packages/mobile/src/navigator/Headers.v2.tsx

```diff
@@ -0,0 +1,117 @@
+import colors from '@celo/react-components/styles/colors.v2'
+import fontStyles from '@celo/react-components/styles/fonts.v2'
+import { componentStyles } from '@celo/react-components/styles/styles'
+import * as React from 'react'
+import { Trans } from 'react-i18next'
+import { StyleSheet, Text, View } from 'react-native'
+import BackButton from 'src/components/BackButton.v2'
+import CancelButton from 'src/components/CancelButton.v2'
+import CurrencyDisplay from 'src/components/CurrencyDisplay'
+import { CURRENCIES, CURRENCY_ENUM } from 'src/geth/consts'
+import i18n, { Namespaces } from 'src/i18n'
+import useSelector from 'src/redux/useSelector'
+import DisconnectBanner from 'src/shared/DisconnectBanner'
+
+export const noHeader = {
+  headerLeft: <View />,
+}
+
+const styles = StyleSheet.create({
+  headerTitle: {
+    ...fontStyles.regular500,
+  },
+  headerSubTitle: {
+    ...fontStyles.small,
+    color: colors.gray4,
+  },
+  header: {
+    alignItems: 'center',
+    justifyContent: 'center',
+  },
+})
+
+export const nuxNavigationOptions = {
+  headerLeftContainerStyle: { paddingHorizontal: 10 },
+  headerLeft: <BackButton />,
+  headerRightContainerStyle: { paddingHorizontal: 10 },
+  headerRight: <View />,
+  headerTitle: <DisconnectBanner />,
+  headerTitleContainerStyle: {
+    alignItems: 'center',
+    flex: 1,
+    justifyContent: 'center',
+  },
+}
+
+export const nuxNavigationOptionsNoBackButton = {
+  ...nuxNavigationOptions,
+  headerLeft: <View />,
+}
+
+export const emptyHeader = {
+  headerTitle: '',
+  headerTitleStyle: [styles.headerTitle, componentStyles.screenHeader],
+  headerTitleContainerStyle: {
+    alignItems: 'center',
+  },
+  headerTitleAlign: 'center',
+}
+
+export const headerWithBackButton = {
+  ...emptyHeader,
+  headerLeft: <BackButton />,
+}
+
+export const headerWithCancelButton = {
+  ...emptyHeader,
+  headerLeft: <CancelButton />,
+}
+
+interface Props {
+  title: string
+  token: CURRENCY_ENUM
+}
+
+export function HeaderTitleWithBalance({ title, token }: Props) {
+  const dollarBalance = useSelector((state) => state.stableToken.balance)
+  const goldBalance = useSelector((state) => state.goldToken.balance)
+
+  const balance = token === CURRENCY_ENUM.GOLD ? goldBalance : dollarBalance
+
+  const subTitle =
+    balance != null ? (
+      <Trans i18nKey="balanceAvailable" ns={Namespaces.global}>
+        <CurrencyDisplay
+          amount={{
+            value: balance,
+            currencyCode: CURRENCIES[token].code,
+          }}
+        />{' '}
+        available
+      </Trans>
+    ) : (
+      // TODO: a null balance doesn't necessarily mean it's loading
```

Comment:
I find the possibility of null balance creates ugly conditionals all across the app and I wonder if we'd be better off just defaulting to 0. We'd lose a signal that says 'I don't know the balance' but it would simplify many places

Reply:
I guess we should move a bit out of the scope of this PR and come with a convenient approach how we represent unknown balance.